ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/LedgerTestCase.scala

```diff
@@ -10,11 +10,12 @@ import scala.concurrent.{ExecutionContext, Future}
 
 final class LedgerTestCase(
     val shortIdentifier: Ref.LedgerString,
+    val name: String,
     val description: String,
     val timeoutScale: Double,
     participants: ParticipantAllocation,
-    runTestCase: Participants => Future[Unit],
+    runTestCase: (Participants, ExecutionContext) => Future[Unit],
 ) {
-  def apply(context: LedgerTestContext)(implicit ec: ExecutionContext): Future[Unit] =
-    context.allocate(participants).flatMap(runTestCase)
+  def run(context: LedgerTestContext)(implicit ec: ExecutionContext): Future[Unit] =
+    context.allocate(participants).flatMap(p => runTestCase(p, ec))
```

Comment:
Perhaps call `runTestCase(ec)(p)` (and define it as `runTestCase: ExecutionContext => Participants => Future[Unit]`)? That way you can define the test case as follows:

```scala
  test(
    "ACSinvalidLedgerId",
    "The ActiveContractService should fail for requests with an invalid ledger identifier",
    allocate(SingleParty),
  ) { implicit ec =>
    case Participants(Participant(ledger, parties @ _*)) =>
      // you no longer need: implicit val e: ExecutionContext = ec
```

Reply:
I have looked for something like this for quite some time, but I have not managed to find a syntax that works for defining the argument of an anonymous function as an implicit. I don't remember if I tried this one, though; I will.

Reply:
I'm probably doing something wrong, and I'd love to be enlightened here. This:
```
    (implicit ec: ExecutionContext) => {
    case Participants(Participant(ledger)) =>
```
gives me this syntax error:
```
error: '}' expected but 'case' found.
```
This yields the same error:
```
    implicit ec => {
    case Participants(Participant(ledger)) =>
```
I can't find any documentation on the implicit keyword. I mean, I can find plenty of documentation that explains how Scala resolves implicits, and that tends to include example snippets of code that use the implicit keywords, but I cannot find an explicit "syntax" documentation that states where the keyword can be used. All the examples I have seen use it in one of three situations: on a class (for implicit conversions via primary constructor), on a named (`def`) method's argument (but then only in the last list of arguments), or in declaring a local `val`.

Reply:
This seems to work when I try it:

```scala
final class IdentityIT extends LedgerTestSuite {
  test(
    "IdNotEmpty",
    "A ledger should return a non-empty string as its identity",
    allocate(NoParties),
  )(implicit ec => {
    case Participants(Participant(ledger)) =>
      Future {
        assert(ledger.ledgerId.nonEmpty, "The returned ledger identifier was empty")
      }
  })
}
```

Curly braces around the implicit also work.

Reply:
Indeed it does. I don't understand what I did differently, but 🤷. I'll update all of them.

Reply:
Thanks!